pkg/secrethub/repo.go

```diff
@@ -14,11 +14,32 @@ type RepoService interface {
 	Delete(path string) error
 	// Get retrieves the repo with the given path.
 	Get(path string) (*api.Repo, error)
+	// EventIterator returns an iterator that retrieves all audit events for a given repo.
+	//
+	// Usage:
+	//  it, err := client.Repos().EventIterator(path)
+	//  if err != nil {
+	//  	   // Handle error
+	//  }
+	//  for {
+	//      event, err := events.Next()
```

Comment:
1. It validates the supplied repo path (as do other client functions that accept a path)
2. Yes :+1: 
3. Sounds good :+1:

Reply:
Update 2 and 3.

For 1, turns out the `EventIterator` also returns errors that might result from evaluating the options that are passed (e.g. from validation of the option, supplying conflicting options, supplying an option twice etc.).
I think it would be confusing to move those errors to the first call of `Next` as well. What do you think?

Reply:
Updated to construct the paginator on first call of `Next`, so that the construction of the iterator does not return an error.